### DISCUSSION TITLE
does this package support fetching values from nested arrays?

### DISCUSSION THREAD

**Discussion by michabbb** [wrk-flow/php-get-typed-value#9] [Q&A]
while reading the readme, i wasn´t able to identify, if it´s possible to get a value from "somewhere" inside an array.
so having the functionality like https://laravel.com/docs/9.x/helpers#method-data-get - would be amazing, in case this
is not (yet) supported ;-)

**Comment by pionl** [comment]
Hi,

Few days ago I wast thinking if I should add the support for it. There are 2 ways I would approach it:

- support dot notation as Laravel (it is easy, but, but it will do "another" check if original key is not in the array).
- support it by passing an array of keys (maybe I prefer it, but harder to type?) $data->getString(['root', 'key']); - it would be more strict

What do you think? Don't wont to support both ways.

Was any part of the readme hard to read? Im not the best at writing :)

Thank you for opening discussion!

**Reply by michabbb** [reply]
hi, one important aspect why I prefer the dot notation, is the possibility to use concatenation with variables. for example 

```php
data_get("first.".$foo.".second.".$bar, $array)
```

and to be honest, yes, I wasn't able to fully understand the example in the Readme in less than 3 minutes, which is usually a personal indicator for me, that very often is part of the decision to decide, if I will invest any more time in something. 

the fact I have build something like this in a super basic way with the opportunity to get the value from any depth of the array, using this package - at least for me - would only, make sense if that feature would be included. 

so that's the main reason, I was asking here ;) 

but don't take this as a immediate feature request. 
if you think that this would be a benefit (for me it is, a huge one), then I guess more people would be interested in this package :)

**Reply by pionl** [reply]
I was planning to make it so, I can make it today. I just wanted to know another "preference".

The concatenation with variables would be nicer with this? :) `$data->getString(['first', $foo, 'second', $bar]);`

**Reply by michabbb** [reply]
in fact it shouldn't make a difference, but as a laravel user who really enjoys that kind of notation (especially when it comes to reading stuff from config files that have large and deep arrays), it's also very "readable" and takes less chars. when getting an api response from a large json it's more readable for me, if I write

```php
$var = data_get('result.products.0.details.price.currency',$response)
```

as using an array. because the dot notation is extremely well readable, like a regular sentence in a regular text.
they array notation needs more brain power and is not that eye friendly.

that's just my opinion 😏

**Reply by pionl** [reply]
Ok, I'm on it 👍

**Reply by michabbb** [reply]
but keep in mind there should be a solution for the "dot in key problem" 🙈

https://laracasts.com/discuss/channels/laravel/keys-with-dots-in-config-files

at this point, the array notation wouldn't make any problems. maybe the optimal way is, to just support both 🤔😏

**Reply by pionl** [reply]
At this time I support only on "root" value with dot.

I've discussed this with colleague and fount it is very complicated to correctly parsing. We could add '\' before '.', but whaat if a key has '\' ... 

We have agreed that only solution to this is to use an array.

**Reply by michabbb** [reply]
your package, your decision 😏 anyway.. if someone wants dots, I guess it should be possible to build a tiny helper function for that....

**Reply by pionl** [reply]
@michabbb sorry I've miss guided you.

We do support dot notation. But edge cases like `co.uk` can be solved only by passing Array. This will ensure that the library behaves always same :)

**Comment by pionl** [comment]
@michabbb Dot notation landed, seems ok to you? I'll tag a release later on. https://php-get-typed-value.wrk-flow.com/#dot-notation

**Reply by michabbb** [reply]
I appreciate your effort for the open-source community here, I am currently not using your package, but as soon as I have a little time, I am happy to test this!  ❤️

**Reply by pionl** [reply]
if you get some time, you can send tips which parts of the docs are bad :) 

Have a good day!

**Reply by michabbb** [reply]
i took a 2nd look into the readme, and with a fresh mind (because it was very late when I discovered your package) I have to say: I guess I do now understand how things work. 🙈 when somethings comes up while implementing your package, I will let you know 😏
